buzzmobile/tests/test_utils/listener.py

```diff
@@ -0,0 +1,28 @@
+#! /usr/bin/env python
+
+"""A ros node that will subscribe to a single message and print it to stdout.
+"""
+import cPickle as pickle
+import sys
+
+import rospy
+
+
+def listen(topic, msg_type):
+    """A function to isolate some of the subscriber logic.
+    """
+    rospy.init_node('mock_listen_'+topic.split('/')[-1], anonymous=True,
+            disable_signals=True)
+    def listener_cb(message):
+        """A default callback for our subscriber.
+        """
+        message.serialize(sys.stdout)
+        rospy.signal_shutdown('message sent')
```

Comment:
yeah im mainly concerned about when, say, on nodeA receiving a single message, it outputting 2 messages in a row (or some variation of that)

But I'll say that until that is a concern, we won't worry about it.

Just make sure to mention in the readme api that check_topic only receives the first message then stops listening